datalad/support/gitrepo.py

```diff
@@ -2760,6 +2760,43 @@ def update_submodule(self, path, mode='checkout', init=False):
             #  an artifact from running submodule update --init manually at
             #  some point, but looking at this code now I worry that it was not
         self._git_custom_command(path, cmd)
+
+        if not init:
+            return
+
+        # track branch originally cloned
+        branch = self.get_active_branch()
+        branch_hexsha = self.get_hexsha(branch)
+        updated_branch = self.get_active_branch()
+        if branch and not updated_branch:
```

Comment:
This is not plausible to me. In the original code block there was a submodule-update in between the two calls to `get_active_branch`. But here - How would `branch` and `updated_branch` ever differ from each other?

Reply:
Arrg, thx for spotting that. Pretty much the entire change was non-sensicle -- pushed an update now.